specklepy/transports/server/server.py

```diff
@@ -59,25 +61,41 @@ def get_object(self, id: str) -> str:
             NotImplementedError,
         )
 
+    def has_objects(self, id_list: List[str]) -> Dict[str, bool]:
+        return {id: False for id in id_list}
+
     def copy_object_and_children(
         self, id: str, target_transport: AbstractTransport
     ) -> str:
-        endpoint = f"{self.url}/objects/{self.stream_id}/{id}"
-        r = self.session.get(endpoint, stream=True)
+        endpoint = f"{self.url}/objects/{self.stream_id}/{id}/single"
+        r = self.session.get(endpoint)
         if r.encoding is None:
             r.encoding = "utf-8"
-        lines = r.iter_lines(decode_unicode=True)
 
-        # save first (root) obj for return
-        root_hash, root_obj = next(lines).split("\t")
-        target_transport.save_object(root_hash, root_obj)
+        root_obj_serialized = r.text
+        root_obj = json.loads(root_obj_serialized)
+        closures = root_obj.get('__closure', {})
+
+        # Check which children are not already in the target transport
+        children_ids = list(closures.keys())
+        children_found_map = target_transport.has_objects(children_ids)
+        new_children_ids = [id for id in children_found_map if not children_found_map[id]]
+
+        # Get the new children
+        endpoint = f"{self.url}/api/getobjects/{self.stream_id}"
+        r = self.session.post(endpoint, data={"objects": json.dumps(new_children_ids)}, stream=True)
+        if r.encoding is None:
+            r.encoding = "utf-8"
+        lines = r.iter_lines(decode_unicode=True)
 
         # iter through returned objects saving them as we go
         for line in lines:
             if line:
                 hash, obj = line.split("\t")
                 target_transport.save_object(hash, obj)
 
+        target_transport.save_object(id, root_obj_serialized)
+
         return root_obj
```

Comment:
ah i've figured out what was causing json decode errors! `copy_object_and_children` used to return a string, but this is not returning a dict. do you need the dict for anything? if not, if this method can return `root_obj_serialized` instead, then we can close #88 . if you need it as a dict, we can alternatively put a condition in the base object serializer

Reply:
done